### ISSUE TITLE
1.68.0 emits STATUS_ILLEGAL_INSTRUCTION on build

### ISSUE THREAD

**Issue by rozensoftware** [rust-lang/rust#108978]
Couldn't build with cargo build.
I tried this code:

```rust
fn main() {
    println!("Hello, world!");
}
```

I expected to see this happen: 

the program should compile fine.

Instead, this happened:

error: could not compile `nanny`                                                                                                                                                                            

Caused by:
  process didn't exit successfully: `rustc --crate-name nanny --edition=2021 src\main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=203 --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=5a2abc74c1c30893 --out-dir D:\Projects\nanny\target\debug\deps -C incremental=D:\Projekty\nanny\target\debug\incremental -L dependency=D:\Projects\nanny\target\debug\deps` (exit code: 0xc000001d, STATUS_ILLEGAL_INSTRUCTION)

`rustc --version --verbose`:
```
rustc 1.68.0 (2c8cc3432 2023-03-06)
binary: rustc
commit-hash: 2c8cc343237b8f7d5a3c3703e3a87f2eb2c54a74
commit-date: 2023-03-06
host: x86_64-pc-windows-msvc
release: 1.68.0
LLVM version: 15.0.6
```

Notes:

cargo check emits the same error.
cargo build --release works correctly.

Having Visual Studio 2022 v17.5.1 installed on my system.

**Comment by workingjubilee** [comment]
That's... weird.
What CPU/Windows OS are you running on?

**Comment by rozensoftware** [comment]
CPU: AMD Phenom(tm) II X4 840 Processor   3.20 GHz
OS: Windows 10 Pro 64bit, 22H2

**Comment by workingjubilee** [comment]
Huh.
Did a prior version of Rust work?

I do know the CPU you are using pushes the edges of Windows 10 support. I would have _thought_ it was still "in the clear", but only barely. I'd ask you to attach a debugger to find where the illegal instruction is invoked but I am not really familiar with the Windows debugging tools.

**Comment by rozensoftware** [comment]
Yes, previous version of Rust did work fine. I have dual booting computer with Linux also. 1.68 compiles correctly mentioned above code on Linux (and other projects also).
I am very sorry, but currently I do not have a ready environment to perform the above test.
I am aware that processor is old. But still, it's a mystery why this happened..

**Comment by workingjubilee** [comment]
So Windows fails, but Linux does not?
I don't know how to explain that...

**Comment by workingjubilee** [comment]
Oh.This might be a consequence of a double-panic...

**Comment by GregGorin** [comment]
I have the same problem. CPU AMD Phenom II X6 1090T

**Comment by KronprinzGit** [comment]
I have the same problem. I used the standalone installer. Core 2 Duo E6320 processor. However, when using "cargo build --release" everything compiles successfully.

**Comment by ehuss** [comment]
Also, if nightly doesn't help, can you try to use a debugger to see where it is faulting? You should be able to launch a debugger on rustc (the path can be shown via `rustup which rustc`), and copy the command-line arguments in. This can be done in Visual Studio if you have that installed.

**Comment by BloodSharp** [comment]
> Could you try on nightly ? Especially the latest one, `nightly-2023-03-14` for rustup, `rustc 1.70.0-nightly (22f247c6f 2023-03-13)` for `rustc -vV`.
> 
> I wonder if it's linked to the ThinLTO miscompilation #109067.

Alright, I installed nightly, by typing:

```
$ rustup install nightly
info: syncing channel updates for 'nightly-x86_64-pc-windows-msvc'
info: latest update on 2023-03-16, rust version 1.70.0-nightly (ab654863c 2023-03-15)
(...)
```

```
$ rustup toolchain list
stable-i686-pc-windows-msvc (default)
stable-x86_64-pc-windows-msvc
nightly-x86_64-pc-windows-msvc
```

```
$ rustup default nightly-x86_64-pc-windows-msvc
info: using existing install for 'nightly-x86_64-pc-windows-msvc'
info: default toolchain set to 'nightly-x86_64-pc-windows-msvc'
```

Then finally both debug and release mode works: 👍
```
$ cargo build
$ cargo build --release
```

I hope this help the people which having the same issue...

**Comment by lqd** [comment]
If bisecting is not possible/easy, checking with another nightly where we know the miscompilation is still present would otherwise be helpful: `rustup install nightly-2023-03-13` and run your builds like so: `cargo +nightly-2023-03-13 build`.

If we're correct, this should fail with the previous `STATUS_ILLEGAL_INSTRUCTION` error.

Thanks in advance.

**Comment by BloodSharp** [comment]
Hello, I installed **caro-bisect-rustc** using nightly x64 msvc and it found a regression...
```
Regression in nightly-2023-03-14
```
The current status is **cloning the rust repository**, I'll let you know once it finish.

**Comment by lqd** [comment]
`--access github` should avoid cloning the repo, but `nightly-2023-03-14` is indeed the one where the change should have happened. Thanks!

**Comment by BloodSharp** [comment]
**cargo-bisect-rustc** has finished! It's safe to remove all the targets and rust.git from my repository folder now?

Regression in 22f247c6f3ed388cb702d01c2ff27da658a8b353

searched nightlies: from nightly-2023-03-01 to nightly-2023-03-16
regressed nightly: nightly-2023-03-14
searched commit range: https://github.com/rust-lang/rust/compare/7b4f48927dce585f747a58083b45ab62b9d73a53...22f247c6f3ed388cb702d01c2ff27da658a8b353
regressed commit: https://github.com/rust-lang/rust/commit/22f247c6f3ed388cb702d01c2ff27da658a8b353

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.5</summary>


Host triple: x86_64-pc-windows-msvc
Reproduce with:
```bash
cargo bisect-rustc 2023-03-01 --end 2023-03-16 --regress success
```
</details>

**Comment by lqd** [comment]
Thanks a bunch! It is indeed safe to remove all these.

**Comment by ehuss** [comment]
@mac198442 Sorry for the miscommunication. This was mentioned on #109067 but not here. A patch release should come out tomorrow. See https://blog.rust-lang.org/inside-rust/2023/03/20/1.68.1-prerelease.html and https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-1681-2023-03-23.

**Comment by mac198442** [comment]
OK just thought my experience was fails on 50% of AMD CPUs I had to test seemed widespread enough to need a backport.

**Comment by mac198442** [comment]
I can verify the my issue is resolved after updating to 1.68.1.

Thanks for the good work.
